gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/EntityServiceAndGatewayServiceConverter.java

```diff
@@ -165,19 +166,18 @@ private static Function convertToQueryFunction(
             .setFunctionName(function.getFunction().name())
             .setAlias(function.getAlias());
 
-    // AVGRATE is adding a specific implementation because Pinot does not directly support this
-    // function
     switch (function.getFunction()) {
       case AVGRATE:
         {
-          builder.setFunctionName(FunctionType.SUM.name()).setAlias(function.getAlias());
-
-          // Adds only the argument that is a column identifier for now.
-          List<org.hypertrace.gateway.service.v1.common.Expression> columns =
-              function.getArgumentsList().stream()
-                  .filter(org.hypertrace.gateway.service.v1.common.Expression::hasColumnIdentifier)
-                  .collect(Collectors.toList());
-          columns.forEach(e -> builder.addArguments(convertToEntityServiceExpression(e)));
+          builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
```

Comment:
My suggestion would be to remove this case entirely (Can just throw, if it doesn't already by default). This logic won't work, because ES doesn't support AVGRATE - and I'm pretty sure it's never used.